kr-core/src/main/clojure/edu/ucdenver/ccp/kr/sparql.clj

```diff
@@ -404,14 +466,23 @@
      (apply str (interleave (map sparql-query-body triple-pattern)
                             ;;(repeat ". \n")))
                             (repeat " \n")))
-   (sparql-operator? (first triple-pattern)) (filter-body triple-pattern)
+   ;; used as a catch-all; this routes anything unusual to a FILTER.
+   ;; (sparql-operator? (first triple-pattern)) (filter-body triple-pattern)
```

Comment:
True and not surprising.  What is the solution, though?  Should we maintain 2 versions, and consider the new one as a sparql 1.1 version?  I remember worrying about this issue, but am not sure how to deal with sparql 1.1 syntax when everything is wearing a filter.  Maybe tomorrow we can talk more about this.

 

From: Marc Daya <notifications@github.com>
Reply-To: UCDenver-ccp/kr <reply@reply.github.com>
Date: Tuesday, May 31, 2016 at 10:50 AM
To: UCDenver-ccp/kr <kr@noreply.github.com>
Cc: Elizabeth White <elizabeth.white@raritanfarm.com>, Author <author@noreply.github.com>
Subject: Re: [UCDenver-ccp/kr] More talking (#4)

 

In kr-core/src/main/clojure/edu/ucdenver/ccp/kr/sparql.clj:
> @@ -404,14 +466,23 @@
>       (apply str (interleave (map sparql-query-body triple-pattern)
>                              ;;(repeat ". \n")))
>                              (repeat " \n")))
> -   (sparql-operator? (first triple-pattern)) (filter-body triple-pattern)
> +   ;; used as a catch-all; this routes anything unusual to a FILTER.
> +   ;; (sparql-operator? (first triple-pattern)) (filter-body triple-pattern)
This change is going to break backwards compatibility. Before operators were implicitly wrapped in a filter, whereas now the filter body must be explicitly specified.

I really should have seen this earlier ):

—
You are receiving this because you authored the thread.
Reply to this email directly, view it on GitHub, or mute the thread.